NPE on shutdown worker too early

**Alluxio Version:**
2.0.0

**Describe the bug**
This is a bug log from https://github.com/Alluxio/alluxio/issues/9279 in 2.0.0-RC2 but the same NPE can still be thrown in 2.0.0 as reported in https://github.com/Alluxio/alluxio/issues/9410

Root cause is invocation an uninitialized [`mSessionCleaner.stop()`](https://github.com/Alluxio/alluxio/blob/branch-2.0/core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java#L263) due to failure in [`start()`](https://github.com/Alluxio/alluxio/blob/branch-2.0/core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java#L199)

```
2019-06-20 13:24:58,254 ERROR ProcessUtils - Uncaught exception while running Alluxio worker, stopping it and exiting.
alluxio.exception.status.UnknownException
	at alluxio.exception.status.AlluxioStatusException.from(AlluxioStatusException.java:154)
	at alluxio.exception.status.AlluxioStatusException.fromStatusRuntimeException(AlluxioStatusException.java:209)
	at alluxio.AbstractClient.retryRPCInternal(AbstractClient.java:371)
	at alluxio.AbstractClient.retryRPC(AbstractClient.java:331)
	at alluxio.worker.block.BlockMasterClient.register(BlockMasterClient.java:222)
	at alluxio.worker.block.BlockMasterSync.registerWithMaster(BlockMasterSync.java:106)
	at alluxio.worker.block.BlockMasterSync.<init>(BlockMasterSync.java:93)
	at alluxio.worker.block.DefaultBlockWorker.start(DefaultBlockWorker.java:214)
	at alluxio.worker.block.DefaultBlockWorker.start(DefaultBlockWorker.java:77)
	at alluxio.Registry.start(Registry.java:131)
	at alluxio.worker.AlluxioWorkerProcess.startWorkers(AlluxioWorkerProcess.java:283)
	at alluxio.worker.AlluxioWorkerProcess.start(AlluxioWorkerProcess.java:235)
	at alluxio.ProcessUtils.run(ProcessUtils.java:35)
	at alluxio.worker.AlluxioWorker.main(AlluxioWorker.java:69)
Caused by: io.grpc.StatusRuntimeException: UNKNOWN
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:233)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:214)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:139)
	at alluxio.grpc.BlockMasterWorkerServiceGrpc$BlockMasterWorkerServiceBlockingStub.registerWorker(BlockMasterWorkerServiceGrpc.java:477)
	at alluxio.worker.block.BlockMasterClient.lambda$register$6(BlockMasterClient.java:223)
	at alluxio.AbstractClient.retryRPCInternal(AbstractClient.java:369)
	... 11 more
2019-06-20 13:24:58,259 INFO  GrpcDataServer - Shutting down RPC Server at 0.0.0.0/0.0.0.0:29999.
2019-06-20 13:25:00,965 INFO  ConfigurationUtils - Alluxio client has loaded configuration from meta master kscpu007.hogpu.cc/10.10.216.8:19998
2019-06-20 13:25:00,965 INFO  ConfigurationUtils - Alluxio client (version 2.0.0-RC3) is trying to load path level configurations
2019-06-20 13:25:00,965 INFO  ConfigurationUtils - Alluxio client has loaded path level configurations
2019-06-20 13:25:01,080 INFO  ConfigurationUtils - Alluxio client has loaded configuration from meta master kscpu007.hogpu.cc/10.10.216.8:19998
2019-06-20 13:25:01,080 INFO  ConfigurationUtils - Alluxio client (version 2.0.0-RC3) is trying to load path level configurations
2019-06-20 13:25:01,080 INFO  ConfigurationUtils - Alluxio client has loaded path level configurations
2019-06-20 13:25:02,677 INFO  GrpcDataServer - Shutting down RPC Server at /usr/local/alluxio/run/d3316fcf-e628-4d40-8190-706dad0ec086.
2019-06-20 13:25:07,100 ERROR ProcessUtils - Uncaught exception while stopping Alluxio worker, simply exiting.
java.lang.NullPointerException
	at alluxio.worker.block.DefaultBlockWorker.stop(DefaultBlockWorker.java:263)
	at alluxio.Registry.stop(Registry.java:148)
	at alluxio.worker.AlluxioWorkerProcess.stopWorkers(AlluxioWorkerProcess.java:287)
	at alluxio.worker.AlluxioWorkerProcess.stop(AlluxioWorkerProcess.java:274)
	at alluxio.ProcessUtils.run(ProcessUtils.java:41)
	at alluxio.worker.AlluxioWorker.main(AlluxioWorker.java:69)
```

**To Reproduce**

**Expected behavior**
No NPE should be thrown ever.

**Urgency**
High

**Additional context**
Please add me as the reviewer

--- a/core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java
+++ b/core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java
@@ -260,7 +260,9 @@ public final class DefaultBlockWorker extends AbstractWorker implements BlockWor
     // 2. Shutdown the executors.
     // 3. Shutdown the clients. This needs to happen after the executors is shutdown because
     //    runnables running in the executors might be using the clients.
-    mSessionCleaner.stop();
+    if (mSessionCleaner != null) {
+      mSessionCleaner.stop();
+    }
     // The executor shutdown needs to be done in a loop with retry because the interrupt
     // signal can sometimes be ignored.
     try {

